javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -186,12 +190,45 @@ protected T getNext() {
         return tabulate(n, ignored -> supplier.get());
     }
 
+    static <T> Tuple2<Iterable<? extends T>, Integer> withSize(Iterable<? extends T> iterable) {
```

Comment:
created all these helpers to bridge the gap between the Java and Javaslang collections.